avro-codegen/src/main/java/com/linkedin/avroutil1/codegen/SpecificRecordClassGenerator.java

```diff
@@ -111,6 +176,1188 @@ protected JavaFileObject generateSpecificFixed(AvroFixedSchema fixedSchema, Spec
+      // read external
+      classBuilder.addField(
+          FieldSpec.builder(DatumReader.class, "READER$", Modifier.PRIVATE, Modifier.STATIC, Modifier.FINAL)
+              .initializer(CodeBlock.of("new $T($L)", SpecificDatumReader.class, "SCHEMA$"))
+              .build());
+
+      MethodSpec.Builder readExternalBuilder = MethodSpec.methodBuilder("readExternal")
+          .addException(IOException.class)
+          .addParameter(java.io.ObjectInput.class, "in")
+          .addModifiers(Modifier.PUBLIC)
+          .addCode(CodeBlock.builder().addStatement("$L.read(this, $T.getDecoder(in))", "READER$", SpecificData.class).build());
+
+      // write external
+      classBuilder.addField(
+          FieldSpec.builder(DatumWriter.class, "WRITER$", Modifier.PRIVATE, Modifier.STATIC, Modifier.FINAL)
+              .initializer(CodeBlock.of("new $T($L)", SpecificDatumWriter.class, "SCHEMA$"))
+              .build());
+
+      MethodSpec.Builder writeExternalBuilder = MethodSpec
+          .methodBuilder("writeExternal")
+          .addException(IOException.class)
+          .addParameter(java.io.ObjectOutput.class, "out")
+          .addModifiers(Modifier.PUBLIC)
+          .addCode(CodeBlock
+              .builder()
+              .addStatement("$L.write(this, $T.getEncoder(out))", "WRITER$", SpecificData.class)
+              .build());
+      readExternalBuilder.addAnnotation(Override.class);
+      writeExternalBuilder.addAnnotation(Override.class);
+
+      classBuilder.addMethod(readExternalBuilder.build());
+      classBuilder.addMethod(writeExternalBuilder.build());
+    }
+
+
+    // add no arg constructor
+    classBuilder.addMethod(MethodSpec.constructorBuilder().addModifiers(Modifier.PUBLIC).build());
+
+
+    if (recordSchema.getFields().size() > 0) {
+
+      // add all arg constructor if #args < 254
+      if(recordSchema.getFields().size() < 254) {
+        MethodSpec.Builder allArgsConstructorBuilder = MethodSpec.constructorBuilder();
+        for (AvroSchemaField field : recordSchema.getFields()) {
+          //if declared schema, use fully qualified class (no import)
+          addFullyQualified(field, config);
+          allArgsConstructorBuilder.addParameter(getParameterSpecForField(field, config))
+              .addStatement("this.$L = $L", field.getName(), field.getName());
+        }
+        classBuilder.addMethod(allArgsConstructorBuilder.build());
+      }
+
+      // Add public/private fields
+      Modifier accessModifier = (config.hasPublicFields())? Modifier.PUBLIC : Modifier.PRIVATE;
+      for (AvroSchemaField field : recordSchema.getFields()) {
+        classBuilder.addField(getFieldSpecBuilder(field, config).addModifiers(accessModifier).build());
+
+        //if declared schema, use fully qualified class (no import)
+        addFullyQualified(field, config);
+
+        //getters
+        classBuilder.addMethod(getGetterMethodSpec(field, config));
```

Comment:
would this cause the getters to show up amongst the fields in the generated code? should we maybe add all fields, then print getters + setters to make the resulting code look better?

Reply:
The generated order is 
SerialId,  SCHEMA$, fields, constructor, getter/setter